boot/freeldr/freeldr/arch/drivers/hwide.c

```diff
@@ -756,7 +783,7 @@ IdentifyDevice(
             {
                 TRACE("Using LBA48 addressing mode.\n");
                 Flags |= ATA_DEVICE_LBA48 | ATA_DEVICE_LBA;
-                TotalSectors = Id.UserAddressableSectors48;
+                TotalSectors = Id.UserAddressableSectors48 + Id.UserAddressableSectors;
```

Comment:
This does not seem right. `UserAddressableSectors48` alone should be sufficient (see e.g. https://elixir.bootlin.com/linux/latest/source/drivers/ata/libata-core.c#L1161 ).
What values do you see in the two struct members that make you think they need to be added?

Reply:
Fixed.

Reply:
Hi, if you say "Original value isn't needed because we use READ command instead of READ CD" then why there is `if (*SectorSize != 0)` and why is this set?
```
*SectorSize = (AtapiCapacity[4] << 24) | (AtapiCapacity[5] << 16) |
                      (AtapiCapacity[6] << 8) | AtapiCapacity[7];
```

Reply:
The zero value tells us that no media has been inserted (See line 771).

Reply:
Isn't this more clear?

At the start:
`*SectorSize = 0`

At that point:
```
if( ((AtapiCapacity[4] << 24) | (AtapiCapacity[5] << 16) | (AtapiCapacity[6] << 8) | AtapiCapacity[7]) != 0) // Check if media is present
	*SectorSize = 2048;
```